src/stdlib/SDL_strtokr.c

```diff
@@ -21,7 +21,7 @@
 #include "SDL_internal.h"
 
 
-char *SDL_strtok_r(char *s1, const char *s2, char **ptr)
+char *SDL_strtok_r(char * SDL_RESTRICT s1, const char * SDL_RESTRICT s2, char ** SDL_RESTRICT ptr)
```

Comment:
I don't think this is correct. A common pattern is to use strtok() on a pointer and fill in the return value with the advanced pointer.

Reply:
It is correct. The memory pointed to by `ptr` should not overlap with either of `s1` or `s2`. Did you look at the pages I linked in the PR description? [Here's musl's implementation of the function if you want another reference](https://github.com/bminor/musl/blob/dd1e63c3638d5f9afb857fccf6ce1415ca5f1b8b/src/string/strtok_r.c).

Reply:
(To clarify, the `restrict` applies to the first level of indirection (the variable to which the pointer to the substring will be written), not the string itself.)